src/query/remote/compressed_codecs.go

```diff
@@ -163,19 +171,28 @@ func compressedSeriesFromSeriesIterator(
 	compressedReplicas := make([]*rpc.M3CompressedValuesReplica, 0, len(replicas))
 	for _, replica := range replicas {
 		replicaSegments := make([]*rpc.M3Segments, 0, len(replicas))
+		replicaBlocks := make([][]xio.BlockReader, 0, len(replicas))
```

Comment:
Rather than allocating these blocks, can we add a `Rewind()` method to `xio.ReaderSliceOfSlicesFromBlockReadersIterator` which just does `func (it *readerSliceOfSlicesIterator) Rewind() { it.idx = -1 }`, and then add a comment to warn users that it will not work if any `.Read()`s have been called on the underlying BlockReaders?

Alternatively if we just straight up add a `BlockReaders() [][]xio.BlockReader` method that will expose these?

Reply:
Good idea. Done